### PR TITLE
GitHub Workflows security hardening

### PR DESCRIPTION
**Pull request by sashashura**
This PR adds explicit [permissions section](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions) to workflows. This is a security best practice because by default workflows run with [extended set of permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) (except from `on: pull_request` [from external forks](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)). By specifying any permission explicitly all others are set to none. By using the principle of least privilege the damage a compromised workflow can do (because of an [injection](https://securitylab.github.com/research/github-actions-untrusted-input/) or compromised third party tool or action) is restricted.
It is recommended to have [most strict permissions on the top level](https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions) and grant write permissions on [job level](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) case by case.

### CODE REVIEW

**Comment by sashashura** [comment]
@francois2metz I have fixed trailing spaces. The build-and-test fails on slack notification step, which looks unrelated to my pr. Please review. Thanks.

File: .github/workflows/auto-merge.yaml [hunk 1]
```diff
@@ -12,8 +12,15 @@ on:
     types:
       - success
 
+permissions:
+  contents: read
+
 jobs:
   automerge:
+    permissions:
```
Comment (francois2metz) [inline .github/workflows/auto-merge.yaml]: We are using a special token here that is not generated by GitHub during the action run. So no need to grant it special permissions. The default one on the top level should have no permissions at all.

File: .github/workflows/on-dev-merge.yaml [hunk 2]
```diff
@@ -3,8 +3,15 @@ on:
...
       - dev
+
+permissions:
+  contents: read
+
 jobs:
   notify-team:
+    permissions:
+      contents: read
+      pull-requests: read
```
Comment (francois2metz) [inline .github/workflows/on-dev-merge.yaml]: I would suggest to either move the pull request permission to the top level or remove the top level permissions.

**Review by francois2metz — COMMENTED** [review]
Hi @sashashura, thanks for the PR. I suggested a few changes. Can you take a look?

File: .github/workflows/auto-merge.yaml (hunk 1 above)
Comment (sashashura) [reply .github/workflows/auto-merge.yaml]: Good catch, missed that.

File: .github/workflows/on-dev-merge.yaml (hunk 2 above)
Comment (sashashura) [reply .github/workflows/on-dev-merge.yaml]: Done!

**Comment by sashashura** [comment]
Done, please take a look.

**Comment by francois2metz** [comment]
This looks good. Could you squash all your commits please?

**Comment by sashashura** [comment]
There was a merge in between, so it was easier to open a new pull request https://github.com/1024pix/pix/pull/4951

**Comment by francois2metz** [comment]
Closing this one then.

**Pull request closed without merging**